packages/pglite-sync/src/index.ts

```diff
@@ -0,0 +1,204 @@
+import type { Extension, PGliteInterface } from '@electric-sql/pglite'
+import { ShapeStream, Message, ChangeMessage } from '@electric-sql/client'
+import type { ShapeStreamOptions } from '@electric-sql/client'
+
+export type MapColumnsMap = Record<string, string>
+export type MapColumnsFn = (message: ChangeMessage<any>) => Record<string, any>
+export type MapColumns = MapColumnsMap | MapColumnsFn
+
+export interface SyncShapeToTableOptions extends ShapeStreamOptions {
+  table: string
+  schema?: string
+  mapColumns?: MapColumns
+  primaryKey: string[]
+}
+
+export interface ElectricSyncOptions {
+  debug?: boolean
+}
+
+async function createPlugin(pg: PGliteInterface, options: ElectricSyncOptions) {
+  const debug = options.debug || false
+  const streams: Array<{
+    stream: ShapeStream
+    aborter: AbortController
+  }> = []
+
+  const namespaceObj = {
+    syncShapeToTable: async (options: SyncShapeToTableOptions) => {
+      const aborter = new AbortController()
+      if (options.signal) {
+        // we new to have our own aborter to be able to abort the stream
+        // but still accept the signal from the user
+        options.signal.addEventListener('abort', () => {
```

Comment:
```suggestion
        options.signal.addEventListener('abort', () => aborter.abort(), { once: true })
```
Can we/should we use a "once" listener here?